### ISSUE TITLE
does percona_server_version support v. 8.0 ?

### ISSUE THREAD

**Issue by arnoldasbet** [Oefenweb/ansible-percona-server#45]
Hi,
does var `percona_server_version` support 8.0 version of percona ? future-proofing here

```
  vars:
    percona_server_version: 8.0
 ```
 
 ```
 TASK [oefenweb.percona_server : version support | check] ***************************************************************************************************************************************************
fatal: [127.0.0.1]: FAILED! => {"changed": false, "msg": "Percona Server version 8.0 is not supported"}
```

```
./vars/main.yml

percona_server_versions_supported:
  - '5.5'
  - '5.6'
  - '5.7'

**Comment by tersmitten** [comment]
No, it does not (yet)

**Comment by zx1986** [comment]
no more (WSREP) cluster support?

**Comment by adepretis** [comment]
When is support for Percona 8 planned?

**Comment by adepretis** [comment]
This is required since Percona decided to move the 8.0 packages to a seperate repositories. Percona server and client packages in the new repository no longer have the "-8.0" suffix.

@arnoldasbet you can make it work by passing role-vars that overload the role-vars , e.g.:

```site.yml
    - name: load | oefenweb.percona_server
      import_role:
        name: oefenweb.percona_server
      vars:
        percona_server_versions_supported:
          - '8.0'

        percona_server_repositories:
          - type: deb
            url: "http://repo.percona.com/ps-80/apt {{ ansible_distribution_release }}"
            component: main
          - type: deb-src
            url: "http://repo.percona.com/ps-80/apt {{ ansible_distribution_release }}"
            component: main
          - type: deb
            url: "http://repo.percona.com/tools/apt {{ ansible_distribution_release }}"
            component: main
          - type: deb-src
            url: "http://repo.percona.com/tools/apt {{ ansible_distribution_release }}"
            component: main

        percona_server_debconf_selections:
          - name: "percona-server-server"
            question: "percona-server-server/root-pass"
            value: "{{ percona_server_root_password }}"
            vtype: password
          - name: "percona-server-server"
            question: "percona-server-server/re-root-pass"
            value: "{{ percona_server_root_password }}"
            vtype: password

        percona_server_dependencies:
          - "percona-server-client"
          - "percona-server-server"
          - "{{ percona_server_libmysqlclient_map[percona_server_version] }}"
          - "python{{ ansible_python_version is version('3', '>=') | ternary('3', '') }}-mysqldb"

        percona_server_libmysqlclient_map:
          8.0: libperconaserverclient21
```

To be safe, there is one thing you have to modify in the role itself though (we do this by automatically applying a patch after `galaxy-install`):

```diff
--- roles/oefenweb.percona_server.orig/tasks/first-run.yml	2021-02-10 16:01:31.000000000 +0100
+++ roles/oefenweb.percona_server/tasks/first-run.yml	2022-09-02 00:11:09.000000000 +0200
@@ -2,7 +2,7 @@
 ---
 - name: first-run | check
   shell: >
-    dpkg-query -W -f='${Status}' percona-server-server-{{ percona_server_version }} | grep -q 'install ok installed'
+    dpkg-query -W -f='${Status}' percona-server-server{{ '' if percona_server_version is version('8', '>=') else '-' + percona_server_version }} | grep -q 'install ok installed'
   register: _mysqld_installed
   check_mode: no
   changed_when: false
```

Would be nice though to have Percona 8+ support built in this role like the other versions. Unfortunately I only had the time to research this workaround and not provider a PR.

**Comment by adepretis** [comment]
Any comments on this @tersmitten?

**Comment by tersmitten** [comment]
Fixed in #47 (finally)
